### PR TITLE
adding controls to ensure region size does not change and sliding stops when the sides are reached

### PR DESCRIPTION
**Pull request by pyros-endgame**
With regards to issue #717 I was looking into it as it was something I wanted to change too so I just went ahead with it. Hopefully that's alright.

Overall, I feel it's much more intuitive now as regions don't magically resize. This is especially true when you set resizing to false but end up dragging the region to the side as well.

### CODE REVIEW

**Comment by katspaugh** [comment]
Awesome, thanks!

File: plugin/wavesurfer.regions.js [hunk 1]
```diff
@@ -424,6 +424,11 @@ WaveSurfer.Region = {
     },
 
     onDrag: function (delta) {
+        var maxEnd = this.wavesurfer.getDuration();
+        if ((this.end + delta) > maxEnd || (this.start + delta) < 0) {
+            delta = 0;
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: If the delta is 0, maybe just return at this point?

**Comment by pyros-endgame** [comment]
Makes sense, I'll fix that.

**Comment by katspaugh** [comment]
Thank you :thumbsup:

**Pull request merged**